lib/ansible/modules/cloud/amazon/ec2_eip.py

```diff
@@ -161,206 +164,303 @@
-from ansible.module_utils.ec2 import HAS_BOTO, ec2_argument_spec, ec2_connect
+    pass  # Handled by AnsibleAWSModule
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    AWSRetry,
+    boto3_conn,
+    ec2_argument_spec,
+    get_aws_connection_info,
+    camel_dict_to_snake_dict,
+    boto3_tag_list_to_ansible_dict,
+    ansible_dict_to_boto3_filter_list,
+    ansible_dict_to_boto3_tag_list,
+    compare_aws_tags
+)
+from ansible.module_utils._text import to_native
+
+DEVICE_TYPE_ENI = 'eni'
+DEVICE_TYPE_INSTANCE = 'instance'
 
 
 class EIPException(Exception):
     pass
 
 
-def associate_ip_and_device(ec2, address, private_ip_address, device_id, allow_reassociation, check_mode, isinstance=True):
-    if address_is_associated_with_device(ec2, address, device_id, isinstance):
-        return {'changed': False}
-
-    # If we're in check mode, nothing else to do
-    if not check_mode:
-        if isinstance:
-            if address.domain == "vpc":
-                res = ec2.associate_address(device_id,
-                                            allocation_id=address.allocation_id,
-                                            private_ip_address=private_ip_address,
-                                            allow_reassociation=allow_reassociation)
-            else:
-                res = ec2.associate_address(device_id,
-                                            public_ip=address.public_ip,
-                                            private_ip_address=private_ip_address,
-                                            allow_reassociation=allow_reassociation)
-        else:
-            res = ec2.associate_address(network_interface_id=device_id,
-                                        allocation_id=address.allocation_id,
-                                        private_ip_address=private_ip_address,
-                                        allow_reassociation=allow_reassociation)
-        if not res:
-            raise EIPException('association failed')
-
-    return {'changed': True}
-
-
-def disassociate_ip_and_device(ec2, address, device_id, check_mode, isinstance=True):
-    if not address_is_associated_with_device(ec2, address, device_id, isinstance):
-        return {'changed': False}
-
-    # If we're in check mode, nothing else to do
-    if not check_mode:
-        if address.domain == 'vpc':
-            res = ec2.disassociate_address(
-                association_id=address.association_id)
-        else:
-            res = ec2.disassociate_address(public_ip=address.public_ip)
-
-        if not res:
-            raise EIPException('disassociation failed')
-
-    return {'changed': True}
-
-
-def _find_address_by_ip(ec2, public_ip):
-    try:
-        return ec2.get_all_addresses([public_ip])[0]
-    except boto.exception.EC2ResponseError as e:
-        if "Address '{}' not found.".format(public_ip) not in e.message:
-            raise
+class AnsibleEc2Eip(object):
```

Comment:
When moving to boto3, we usually try to minimize the amount of code change to just altering the API calls/parameter names that are different between boto/boto3. Moving code into a class makes it very hard to review while also checking the boto3 compatibility.